content/blog/2020/how-selenium-works-2-navigation.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "How Selenium Works: Episode 2 - Navigation"
+publishDate: 2020-02-13
+---
+
+After an interaction on the last weekend of January 2020, on a Selenium Issue where someone said “why can’t you just…” after I explained the issue I thought that I would start explaining commands in Selenium WebDriver and why we landed on the design that we have today.
+
+I will repeat this on every page of the series but a lot, an annoying amount sometimes, of thinking goes into how every little bit of Selenium works. 
+
+In this episode we are going to look into the sheer amount of work that goes into navigation. 
+
+{{< highlight python >}}
+from selenium import webdriver
+
+driver = webdriver.Firefox()
+driver.get("https://www.theautomatedtester.co.uk")
+{{< /highlight >}}
+
+We can see above... it looks easy enough right... no!
+
+Actually, it leads me to a pet peeve in interviews. If someone ever asks you to ever describe what happens in a browser when you type a URL and press enter there is a high probability that they have no real idea what goes into navigation. Anyway... back to Selenium and how we navigate.
+
+# `driver.get`
+
+Once we send the request over the [transportation layer]({{< relref "/blog/how-selenium-works-transport">}}) the real fun starts. We need to figure out where we want to go and tell the browser to start going there for us. Unfortunately this will lead to the first of the many problems for automators.
+
+## Certificates
+
+If you have ever worked in major corporate companies where technology is not the core then you will understand the pain that a development team might have to endure. Now double that pain and you're starting to get an idea of the pain that testers and automators have to deal with.
+
+One of these pains is certificates. Companies are cheap and will do self-signed certificates and other monstrosities. Especially in the early days of Selenium when there wasn't services like [Let's Encrypt](https://letsencrypt.org/). And even now, most Developers and QA teams rarely have access to change configurations on their test environments or on their CI servers. We needed to come up with a way to circumvent the certificates. (_This is one of the first reasons why Selenium is seen as a security risk_)
```

Comment:
Nit: "weren't" reads better than "wasn't". I'd also be tempted to have `monstrosities, especially`.